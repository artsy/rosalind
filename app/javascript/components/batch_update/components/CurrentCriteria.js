import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { avantGarde } from './Layout'
import {
  SelectedCreatedAfterDate,
  SelectedCreatedBeforeDate,
  SelectedGene,
  SelectedKeyword,
  SelectedTag,
  SelectedArtist,
  SelectedPartner,
  SelectedFair,
  SelectedPrices,
  SelectedAttributionClass,
  SelectedRestrictedArtworkIDs,
} from './Selected'

function CurrentCriteria(props) {
  const {
    artists,
    attributionClass,
    className,
    clearState,
    createdAfterDate,
    createdBeforeDate,
    fair,
    genes,
    keywords,
    onRemoveKeyword,
    onRemoveGene,
    onRemoveTag,
    onRemoveArtist,
    partner,
    restrictedArtworkIDs,
    tags,
    minPrice,
    maxPrice,
  } = props

  return (
    <div className={className}>
      {keywords.length > 0 && (
        <CurrentKeywords
          keywords={keywords}
          onRemoveKeyword={onRemoveKeyword}
        />
      )}
      {restrictedArtworkIDs.length > 0 && (
        <SelectedRestrictedArtworkIDs
          artworkIDs={restrictedArtworkIDs}
          onRemove={clearState}
        />
      )}
      {genes.length > 0 && (
        <CurrentGenes genes={genes} onRemoveGene={onRemoveGene} />
      )}
      {tags.length > 0 && <CurrentTags tags={tags} onRemoveTag={onRemoveTag} />}
      {artists.length > 0 && (
        <CurrentArtists artists={artists} onRemoveArtist={onRemoveArtist} />
      )}
      {partner && (
        <SelectedPartner name={partner.name} clearState={clearState} />
      )}
      {fair && <SelectedFair name={fair.name} clearState={clearState} />}
      {attributionClass && (
        <SelectedAttributionClass
          name={attributionClass.name}
          clearState={clearState}
        />
      )}
      {createdAfterDate && (
        <SelectedCreatedAfterDate
          name={createdAfterDate}
          clearState={clearState}
        />
      )}
      {createdBeforeDate && (
        <SelectedCreatedBeforeDate
          name={createdBeforeDate}
          clearState={clearState}
        />
      )}
      {(minPrice !== null || maxPrice !== null) && (
        <SelectedPrices
          minPrice={minPrice}
          maxPrice={maxPrice}
          clearState={clearState}
        />
      )}
    </div>
  )
}

CurrentCriteria.propTypes = {
  artists: PropTypes.arrayOf(PropTypes.object),
  clearState: PropTypes.func,
  createdAfterDate: PropTypes.string,
  createdBeforeDate: PropTypes.string,
  fair: PropTypes.object,
  genes: PropTypes.arrayOf(PropTypes.object),
  onRemoveGene: PropTypes.func.isRequired,
  onRemoveTag: PropTypes.func.isRequired,
  partner: PropTypes.object,
  tags: PropTypes.arrayOf(PropTypes.object),
}

function CurrentGenes(props) {
  const { genes, onRemoveGene } = props
  return (
    <div>
      <h2>Genes</h2>
      {genes.map(g => (
        <SelectedGene key={g.id} name={g.name} onRemove={onRemoveGene} />
      ))}
    </div>
  )
}

function CurrentKeywords(props) {
  const { keywords, onRemoveKeyword } = props
  return (
    <div>
      <h2>Keywords</h2>
      {keywords.map(k => (
        <SelectedKeyword key={k} text={k} onRemove={onRemoveKeyword} />
      ))}
    </div>
  )
}

function CurrentTags(props) {
  const { tags, onRemoveTag } = props
  return (
    <div>
      <h2>Tags</h2>
      {tags.map(t => (
        <SelectedTag key={t.id} name={t.name} onRemove={onRemoveTag} />
      ))}
    </div>
  )
}

function CurrentArtists(props) {
  const { artists, onRemoveArtist } = props
  return (
    <div>
      <h2>Artists</h2>
      {artists.map(a => (
        <SelectedArtist
          key={a.id}
          id={a.id}
          name={a.name}
          onRemove={onRemoveArtist}
        />
      ))}
    </div>
  )
}

/* default styled component */

const StyledCurrentCriteria = styled(CurrentCriteria)`
  h2 {
    ${avantGarde}
    color: #999;
    font-size: 0.75em;
    font-weight: 400;
    letter-spacing: 0.1em;
    margin-top: 1.5em;
    text-transform: uppercase;
  }
`

export default StyledCurrentCriteria
