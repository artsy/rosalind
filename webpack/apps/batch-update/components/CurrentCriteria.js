import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { avantGarde } from './Layout'
import {
  SelectedCreatedAfterDate,
  SelectedCreatedBeforeDate,
  SelectedGene,
  SelectedTag,
  SelectedPartner,
  SelectedFair
} from './Selected'

function CurrentCriteria (props) {
  const {
    className,
    clearState,
    createdAfterDate,
    createdBeforeDate,
    fair,
    genes,
    onRemoveGene,
    onRemoveTag,
    partner,
    tags
  } = props

  return (
    <div className={className}>
      {genes.length > 0 && <CurrentGenes genes={genes} onRemoveGene={onRemoveGene} />}
      {tags.length > 0 && <CurrentTags tags={tags} onRemoveTag={onRemoveTag} />}
      {partner && <SelectedPartner name={partner.name} clearState={clearState} />}
      {fair && <SelectedFair name={fair.name} clearState={clearState} />}
      {createdAfterDate && <SelectedCreatedAfterDate name={createdAfterDate} clearState={clearState} />}
      {createdBeforeDate && <SelectedCreatedBeforeDate name={createdBeforeDate} clearState={clearState} />}
    </div>
  )
}

CurrentCriteria.propTypes = {
  clearState: PropTypes.func,
  createdAfterDate: PropTypes.string,
  createdBeforeDate: PropTypes.string,
  fair: PropTypes.object,
  genes: PropTypes.arrayOf(PropTypes.object),
  onRemoveGene: PropTypes.func.isRequired,
  onRemoveTag: PropTypes.func.isRequired,
  partner: PropTypes.object,
  tags: PropTypes.arrayOf(PropTypes.object)
}

function CurrentGenes (props) {
  const { genes, onRemoveGene } = props
  return (
    <div>
      <h2>Genes</h2>
      {genes.map(g => <SelectedGene key={g.id} name={g.name} onRemove={onRemoveGene} />)}
    </div>
  )
}

function CurrentTags (props) {
  const { tags, onRemoveTag } = props
  return (
    <div>
      <h2>Tags</h2>
      {tags.map(t => <SelectedTag key={t.id} name={t.name} onRemove={onRemoveTag} />)}
    </div>
  )
}

/* default styled component */

const StyledCurrentCriteria = styled(CurrentCriteria)`
  h2 {
    ${avantGarde}
    color: #999;
    font-size: 0.75em;
    font-weight: 400;
    letter-spacing: 0.1em;
    margin-top: 1.5em;
    text-transform: uppercase;
  }
`

export default StyledCurrentCriteria
