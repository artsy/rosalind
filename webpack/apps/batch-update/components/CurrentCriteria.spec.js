import React from 'react'
import renderer from 'react-test-renderer'
import CurrentCriteria from './CurrentCriteria'

let props

beforeEach(() => {
  props = {
    genes: [],
    tags: [],
    partner: null,
    fair: null,
    onRemoveGene: jest.fn(),
    onRemoveTag: jest.fn(),
    onClearPartner: jest.fn(),
    onClearFair: jest.fn()
  }
})

it('renders nothing if there are no selected criteria', () => {
  const rendered = renderer.create(<CurrentCriteria {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders the selected genes', () => {
  props.genes = [
    {id: 'foo', name: 'Foo'},
    {id: 'bar', name: 'Bar'}
  ]
  const rendered = renderer.create(<CurrentCriteria {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders the selected tags', () => {
  props.tags = [
    {id: 'foo', name: 'Foo'},
    {id: 'bar', name: 'Bar'}
  ]
  const rendered = renderer.create(<CurrentCriteria {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders the selected partner', () => {
  props.partner = {id: 'foo', name: 'Gallery Foo'}
  const rendered = renderer.create(<CurrentCriteria {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders the selected fair', () => {
  props.fair = {id: 'foo', name: 'FooFair'}
  const rendered = renderer.create(<CurrentCriteria {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})
