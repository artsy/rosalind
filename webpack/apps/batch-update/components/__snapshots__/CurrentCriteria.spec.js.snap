// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nothing if there are no selected criteria 1`] = `
<div
  className="cMTqqq"
/>
`;

exports[`renders the selected fair 1`] = `
<div
  className="cMTqqq"
>
  <div>
    <h2>
      Fair
    </h2>
    <div
      className="crrNnH"
    >
      FooFair
      <a
        className="remove"
        href="#"
        onClick={[Function]}
      >
        ✕
      </a>
    </div>
  </div>
</div>
`;

exports[`renders the selected genes 1`] = `
<div
  className="cMTqqq"
>
  <div>
    <h2>
      Genes
    </h2>
    <div
      className="crrNnH"
    >
      Foo
      <a
        className="remove"
        href="#"
        onClick={[Function]}
      >
        ✕
      </a>
    </div>
    <div
      className="crrNnH"
    >
      Bar
      <a
        className="remove"
        href="#"
        onClick={[Function]}
      >
        ✕
      </a>
    </div>
  </div>
</div>
`;

exports[`renders the selected partner 1`] = `
<div
  className="cMTqqq"
>
  <div>
    <h2>
      Partner
    </h2>
    <div
      className="crrNnH"
    >
      Gallery Foo
      <a
        className="remove"
        href="#"
        onClick={[Function]}
      >
        ✕
      </a>
    </div>
  </div>
</div>
`;

exports[`renders the selected tags 1`] = `
<div
  className="cMTqqq"
>
  <div>
    <h2>
      Tags
    </h2>
    <div
      className="crrNnH"
    >
      Foo
      <a
        className="remove"
        href="#"
        onClick={[Function]}
      >
        ✕
      </a>
    </div>
    <div
      className="crrNnH"
    >
      Bar
      <a
        className="remove"
        href="#"
        onClick={[Function]}
      >
        ✕
      </a>
    </div>
  </div>
</div>
`;
